packages/m2m-gateway/test/api/purposes/createPurpose.test.ts

```diff
@@ -0,0 +1,117 @@
+
+  const mockPurposeSeed: m2mGatewayApi.PurposeSeed = {
+    dailyCalls: mockPurpose.data.versions[0].dailyCalls,
+    description: mockPurpose.data.description,
+    eserviceId: mockPurpose.data.eserviceId,
+    isFreeOfCharge: mockPurpose.data.isFreeOfCharge,
+    title: mockPurpose.data.title,
+  };
+
+  const mockM2MPurpose: m2mGatewayApi.Purpose = toM2MGatewayApiPurpose({
+    purpose: mockPurpose.data,
+    logger: genericLogger,
+  });
+
+  const makeRequest = async (token: string, body: m2mGatewayApi.PurposeSeed) =>
+    request(api)
+      .post(`${appBasePath}/purposes`)
+      .set("Authorization", `Bearer ${token}`)
+      .send(body);
+
+  const authorizedRoles: AuthRole[] = [authRole.M2M_ADMIN_ROLE];
+  it.each(authorizedRoles)(
+    "Should return 200 and perform service calls for user with role %s",
+    async (role) => {
+      mockPurposeService.createPurpose = vi
+        .fn()
+        .mockResolvedValue(mockM2MPurpose);
+
+      const token = generateToken(role);
+      const res = await makeRequest(token, mockPurposeSeed);
+
+      expect(res.status).toBe(201);
+      expect(res.body).toEqual(mockM2MPurpose);
+    }
+  );
+
+  it.each(
+    Object.values(authRole).filter((role) => !authorizedRoles.includes(role))
+  )("Should return 403 for user with role %s", async (role) => {
+    const token = generateToken(role);
+    const res = await makeRequest(token, mockPurposeSeed);
+    expect(res.status).toBe(403);
+  });
+
+  it.each([
+    { invalidParam: "invalidValue" },
+    { ...mockPurposeSeed, extraParam: -1 },
+    { ...mockPurposeSeed, description: "short" },
+  ])("Should return 400 if passed invalid delegation seed", async (body) => {
+    const token = generateToken(authRole.M2M_ADMIN_ROLE);
+    const res = await makeRequest(
+      token,
+      body as unknown as m2mGatewayApi.PurposeSeed
+    );
+
+    expect(res.status).toBe(400);
+  });
+
+  it("Should return 500 in case of missingMetadata error", async () => {
+    mockPurposeService.createPurpose = vi
+      .fn()
+      .mockRejectedValue(missingMetadata());
+    const token = generateToken(authRole.M2M_ADMIN_ROLE);
+    const res = await makeRequest(token, mockPurposeSeed);
+
+    expect(res.status).toBe(500);
+  });
+
+  it("Should return 500 in case of resourcePollingTimeout error", async () => {
+    mockPurposeService.createPurpose = vi
+      .fn()
+      .mockRejectedValue(resourcePollingTimeout(3));
+    const token = generateToken(authRole.M2M_ADMIN_ROLE);
+    const res = await makeRequest(token, mockPurposeSeed);
+
+    expect(res.status).toBe(500);
+  });
+
+  it("Should return 500 in case of purposeNotFound error", async () => {
+    mockPurposeService.createPurpose = vi
+      .fn()
+      .mockRejectedValue(purposeNotFound(mockPurpose.data.id));
+    const token = generateToken(authRole.M2M_ADMIN_ROLE);
+    const res = await makeRequest(token, mockPurposeSeed);
+
+    expect(res.status).toBe(500);
+  });
```

Comment:
This is probably not needed because on create we never catch and rethrow a purposeNotFound error
```suggestion
```